crates/telio-wg/src/wg.rs

```
@@ -544,12 +630,39 @@ impl State {
             } else {
                 return false;
             };
-            if !try_send(PeerState::Connecting, peer.clone()).await {
+
+            // Node is new and default LinkState is down. Save it before sending the event
+            self.no_iink_detection_info
+                .insert(*key, NoLinkDetectionInfo::default());
+            if !self
```

These `if`s are a bit strange, can we return an error with `?` instead?